lib/Doctrine/Migrations/Tools/Console/ConsoleInputMigratorConfigurationFactory.php

```diff
@@ -21,12 +21,26 @@ public function getMigratorConfiguration(InputInterface $input): MigratorConfigu
     {
         $timeAllQueries = $input->hasOption('query-time') ? (bool) $input->getOption('query-time') : false;
         $dryRun         = $input->hasOption('dry-run') ? (bool) $input->getOption('dry-run') : false;
-        $allOrNothing   = $input->hasOption('all-or-nothing') ? $input->getOption('all-or-nothing') : null;
-        $allOrNothing   = (bool) ($allOrNothing ?? $this->configuration->isAllOrNothing());
+        $allOrNothing   = $this->determineAllOrNothingValueFrom($input) ?? $this->configuration->isAllOrNothing();
 
         return (new MigratorConfiguration())
             ->setDryRun($dryRun)
             ->setTimeAllQueries($timeAllQueries)
             ->setAllOrNothing($allOrNothing);
     }
+
+    private function determineAllOrNothingValueFrom(InputInterface $input): ?bool
+    {
+        $allOrNothingOption = null;
+
+        if ($input->hasOption('all-or-nothing')) {
+            $allOrNothingOption = $input->getOption('all-or-nothing');
+        }
+
+        if ($allOrNothingOption === 'notprovided') {
+            return null;
+        }
+
+        return (bool) ($allOrNothingOption ?? true);
```

Comment:
If the command does not have the option, the value will default to `true`, but I guess it does not matter since `all-or-nothing` is not used when the option is not defined.

Reply:
This is a problem, the default value of `MigrationConfiguration` property `allOrNothing` is `false`, these changes always set the value to true if the option isn't passed at all.

Reply:
You mean the option is used although it's not defined in the command? Where?

Reply:
The option itself isn't used, but [ConsoleInputMigratorConfigurationFactory->getMigratorConfiguration()](https://github.com/doctrine/migrations/blob/3.5.x/lib/Doctrine/Migrations/Tools/Console/ConsoleInputMigratorConfigurationFactory.php#L20) is being called in [`ExecuteCommand->execute()`](https://github.com/doctrine/migrations/blob/3.5.x/lib/Doctrine/Migrations/Tools/Console/Command/ExecuteCommand.php#L115) which in turn creates a `MigratorConfiguration` with `allOrNothing` set to `true` because `ExecuteCommand` doesn't have an `all-or-nothing` option which results in the value not defaulting to `notprovided`.

Reply:
Basically this method should look like this:
```
        if (!$input->hasOption('all-or-nothing')) {
            return null;
        }

        $allOrNothingOption = $input->getOption('all-or-nothing');

        if ($allOrNothingOption === 'notprovided') {
            return null;
        }

        return (bool) ($allOrNothingOption ?? true);
```

Reply:
Created a PR with the fix: https://github.com/doctrine/migrations/pull/1308